feat(Integrationtest): integration test param support

abhinandan13jan opened the pull request:
## Fixes 
<!-- For e.g Fixes: https://issues.redhat.com/browse/HAC-XXX -->
https://issues.redhat.com/browse/HAC-4596

## Description
<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->
- Adds parameter support for IntegrationTest
- Creates a Param component

## Type of change
<!-- Please delete options that are not relevant. -->

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
<!-- If change affects UI in any way, tag relevant UX people and add screenshots/gifs  -->

<img width="829" alt="Screenshot 2023-09-13 at 9 26 52 PM" src="https://github.com/openshift/hac-dev/assets/24852534/7fb1dc3b-dd4c-4a61-b788-07835144064e">
<img width="835" alt="Screenshot 2023-09-13 at 9 52 22 PM" src="https://github.com/openshift/hac-dev/assets/24852534/33a0745e-b7b1-4f47-896f-31621bb55989">
<img width="822" alt="Screenshot 2023-09-13 at 9 54 32 PM" src="https://github.com/openshift/hac-dev/assets/24852534/e403d58a-0fc3-4ee9-a6b3-22b10b0a601b">
<img width="844" alt="Screenshot 2023-09-13 at 9 54 43 PM" src="https://github.com/openshift/hac-dev/assets/24852534/571ce984-db30-47cb-bdc0-af2e88a6f683">
<img width="771" alt="Screenshot 2023-09-13 at 9 54 57 PM" src="https://github.com/openshift/hac-dev/assets/24852534/fa349c66-cda0-458e-a1fe-8339a076d31a">

## How to test or reproduce?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

<!-- - [ ] Test A -->
<!-- - [ ] Test B -->

<!-- **Test Configuration(s)**: -->


## Browser conformance: 
<!-- To mark tested browsers, use [x] -->
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge


<!-- ## Checklist: -->

<!-- 
- [ ] Code follows the style guidelines
- [ ] Self-reviewed the code
- [ ] Added comments in hard-to-understand areas
- [ ] Made corresponding changes to the documentation
- [ ] Changes generate no new warnings
- [ ] Added tests that prove this fix is effective or that the feature works
- [ ] New and existing unit tests pass locally with new changes
- [ ] Any dependent changes have been merged and published in downstream modules 
-->

MariaLeonova commented:
@abhinandan13jan this looks great! I just have one question: why is there a grey outline around the parameters field?

Misjohns commented:
There's a few elements that look off. I've included a redline with recommendations. Please refer to this demo for padding and text styling: https://www.patternfly.org/components/forms/form
<img width="1445" alt="image" src="https://github.com/openshift/hac-dev/assets/97301019/710de759-efd5-421c-91dc-26dd2a5d3c08">

abhinandan13jan commented:
@MariaLeonova By default, the expandable card has an outline, removed it.

<img width="811" alt="Screenshot 2023-09-13 at 6 28 53 PM" src="https://github.com/openshift/hac-dev/assets/24852534/468a40e5-9fc4-4c50-8f74-d3c0f6d6764b">

@Misjohns 
1. Updated the spacing. Was previously using the default spacing of patternfly Data-List
2. Removed the bold weight from `Parameter 1` name
3. The `Parameters` font size is exactly the same as other fields in the `IntegrationTestForm`. If we make it larger will look different from other fields like `name` `git url`. Do let me know if it needs to be updated

<img width="865" alt="Screenshot 2023-09-13 at 7 09 20 PM" src="https://github.com/openshift/hac-dev/assets/24852534/cc43ad98-8922-4836-83b6-04855b42b997">

Misjohns commented:
The expandable section **Parameters** isn't a field label so it should be same size as the "Mark as optional..."
https://www.patternfly.org/components/expandable-section

abhinandan13jan commented:
> The expandable section **Parameters** isn't a field label so it should be same size as the "Mark as optional..." https://www.patternfly.org/components/expandable-section

@Misjohns updated
<img width="835" alt="Screenshot 2023-09-13 at 9 52 22 PM" src="https://github.com/openshift/hac-dev/assets/24852534/71ba07db-38b9-4a70-80d7-14eb50bb693d">
<img width="822" alt="Screenshot 2023-09-13 at 9 54 32 PM" src="https://github.com/openshift/hac-dev/assets/24852534/9d13d52f-b5d2-407c-bf2c-ef69a3885b5b">
<img width="771" alt="Screenshot 2023-09-13 at 9 54 57 PM" src="https://github.com/openshift/hac-dev/assets/24852534/ac5ee684-b10e-4a27-b24e-b70520a5be46">

Misjohns commented:
LGTM!

rohitkrai03 commented on src/components/IntegrationTest/IntegrationTestParams.tsx:
```diff
@@ -0,0 +1,323 @@
+    }
+  };
+
+  const toggleExpandedState = (i) => {
```
Isn't there a uncontrolled version of expand collapse section which handles the state automatically?

abhinandan13jan replied:
I am not sure, but I also need to expand new params when created.

rohitkrai03 commented on src/components/IntegrationTest/IntegrationTestParams.tsx:
```diff
@@ -0,0 +1,323 @@
+        </CardHeader>
+        <CardExpandableContent>
+          <CardBody className="review-component-card__card-body">
+            {Array.isArray(parameters) && parameters.length > 0 ? (
```
Wouldn't it be better and easier to handle this using `FieldArray` by Formik?

karthikjeeyar commented on src/components/IntegrationTest/FormikParamsField.tsx:
```diff
@@ -0,0 +1,259 @@
+                                      width={3}
+                                      className="pf-v5-u-pl-xl pf-v5-u-pt-0"
+                                    >
+                                      <FormGroup label="Name" isRequired>
```
Name and value field is optional in design doc.

```suggestion
                                      <FormGroup label="Name">
```
<img width="856" alt="image" src="https://github.com/openshift/hac-dev/assets/9964343/0905dcf9-2493-4ff3-a1c1-c963123f3196">

karthikjeeyar reviewed (commented):
While creating the integration test we should not set the param value if it is empty. Right now if you click on add value multiple times and create a integration test then it sets empty values in spec.params as you can see here below in the Edit integration test flow.

<img width="752" alt="editIntegraionTest" src="https://github.com/openshift/hac-dev/assets/9964343/f8d6aff3-af26-4ed2-9502-2992e3f20836">

karthikjeeyar commented:
Missing Help tooltip which is outlined in the design

 design doc:
<img width="601" alt="Screenshot 2023-09-27 at 1 53 22 PM" src="https://github.com/openshift/hac-dev/assets/9964343/eb5921c5-ff32-475c-9911-536c96c8f23a">

abhinandan13jan commented:
> Missing Help tooltip which is outlined in the design
> 
> design doc: <img alt="Screenshot 2023-09-27 at 1 53 22 PM" width="601" src="https://user-images.githubusercontent.com/9964343/270912982-eb5921c5-ff32-475c-9911-536c96c8f23a.png">

So we decided in the meeting, not to use a tooltip, as this is a straightforward field and we dont have tooltips for any other fields.

karthikjeeyar commented:
/approve
/lgtm
